.github/workflows/static_checks.yaml

```diff
@@ -0,0 +1,49 @@
+name: Static checks
+
+concurrency:
+  group: static-checks-${{ github.event_name }}-${{ github.sha }}
+  cancel-in-progress: true
+
+on:
+  workflow_call:
+    inputs:
+      runner:
+        required: true
+        type: string
+      builder:
+        required: true
+        type: string
+      ct-matrix:
+        required: true
+        type: string
+
+env:
+  IS_CI: "yes"
+
+jobs:
+  static_checks:
+    runs-on: ${{ inputs.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(inputs.ct-matrix) }}
+    container: "ghcr.io/emqx/emqx-builder/${{ matrix.builder }}:${{ matrix.elixir }}-${{ matrix.otp }}-ubuntu22.04"
+    steps:
+      - uses: AutoModality/action-clean@v1
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ matrix.profile }}
+      - name: extract artifact
+        run: |
+          unzip -o -q ${{ matrix.profile }}.zip
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+      - uses: actions/cache@v3
+        with:
+          path: "emqx_dialyzer_${{ matrix.otp }}_plt"
+          key: rebar3-dialyzer-plt-${{ matrix.profile }}-${{ matrix.otp }}-${{ hashFiles('rebar.*', 'apps/*/rebar.*', 'lib-ee/*/rebar.*') }}
+          restore-keys: |
+            rebar3-dialyzer-plt-${{ matrix.profile }}-${{ matrix.otp }}-
+      - name: run static checks
+        env:
+          PROFILE: ${{ matrix.profile }}
+        run: make static_checks
```

Comment:
I noticed that `scripts/check-i18n-style.escript` apparently never exits with an error exit code, so it seems like that check is being ignored in practice.  :thinking: 

(not introduced by this PR)